File: TestGenerationTool/mocks/Cloud.py
class Cloud:

    def __init__(self):
        self.valuesToDiscard = 0
        self.threshold = 160
        self.numReadings = 0
        self.maxReadings = 20
        self.glucoseReadings = [0] * self.maxReadings

    def getCriticalCount(self):
        res = 0
        for x in self.glucoseReadings:
            if(x > self.threshold):
                res += 1
        return res

    def reset(self):
        self.valuesToDiscard = 0;
        self.numReadings = 0;
        self.glucoseReadings = [0] * self.maxReadings;

    def receiveOver(self):
        if(self.valuesToDiscard == 0):
            self.glucoseReadings[self.numReadings] = self.threshold+1
            self.numReadings += 1
            if(self.numReadings == 21):
                self.numReadings = 1
        else:
            self.valuesToDiscard -= 1

    def receiveUnder(self):
        if (self.valuesToDiscard == 0):
            self.glucoseReadings[self.numReadings] = self.threshold - 1
            self.numReadings += 1
            if (self.numReadings == 21):
                self.numReadings = 1
        else:
            self.valuesToDiscard -= 1

    def resetReadings(self):
        self.numReadings = 0;
        self.glucoseReadings = [0] * self.maxReadings;

    def discardNext(self, qty):
        self.valuesToDiscard = qty;

    def getThreshold(self):
        return self.threshold

    def getNumReadings(self):
        return self.numReadings

    def getMaxReadings(self):
        return self.maxReadings

File: GeneratedTests/DiaMHTests2_NewModel/src/main/java/DiaMHTestsMaven/wrappers/Cloud.java
package DiaMHTestsMaven.wrappers;
import org.eclipse.paho.client.mqttv3.MqttClient;
import org.eclipse.paho.client.mqttv3.MqttConnectOptions;
import org.eclipse.paho.client.mqttv3.MqttException;
import org.eclipse.paho.client.mqttv3.MqttMessage;
import org.eclipse.paho.client.mqttv3.persist.MemoryPersistence;

public class Cloud implements ICloud {
	private static final String CLOUD_RESET = "diamh/cloud/reset";
	private static final String MQTT_BROKER = "tcp://mqtt.eclipse.org:1883";
	
	private int valuesToDiscard;
	private int threshold;
	private int numReadings;
	private int maxReadings;
	private int[] glucoseReadings;
	
	
	
	public Cloud() {
        valuesToDiscard = 0;
        threshold = 160;
        numReadings = 0;
        maxReadings = 20;
        glucoseReadings = new int[maxReadings];
	}
	
	@Override
	public void reset() {
		try {
			String clientId = MqttClient.generateClientId();

	        System.out.println("Connecting to broker: "+MQTT_BROKER);
        	MqttConnectOptions conOpt = new MqttConnectOptions();
        	conOpt.setCleanSession(true);
            MqttClient sampleClient = new MqttClient(MQTT_BROKER, clientId, new MemoryPersistence());
            sampleClient.connect(conOpt);
	        System.out.println("Connected");
	        
			MqttMessage message = new MqttMessage("1".getBytes());
	        message.setQos(1);
	        sampleClient.publish(CLOUD_RESET, message);
		}catch(MqttException me) {
            System.out.println("reason "+me.getReasonCode());
            System.out.println("msg "+me.getMessage());
            System.out.println("loc "+me.getLocalizedMessage());
            System.out.println("cause "+me.getCause());
            System.out.println("excep "+me);
            me.printStackTrace();
        } finally {
            valuesToDiscard = 0;
            threshold = 160;
            numReadings = 0;
            maxReadings = 20;
            glucoseReadings = new int[maxReadings];
        }
	}


	@Override
	public int getCriticalCount() {
		int count = 0;
		for(int i: glucoseReadings) {
			if(i>threshold) count++;
		}
		return count;
	}


	@Override
	public void receiveUnder() {
		if(valuesToDiscard == 0) {
			glucoseReadings[numReadings] = threshold-1;
            numReadings++;
		}
        else {
            valuesToDiscard--;
        }
	}


	@Override
	public void receiveOver() {
		if(valuesToDiscard == 0) {
			glucoseReadings[numReadings] = threshold+1;
            numReadings++;
		}
        else {
            valuesToDiscard--;
        }
		
	}


	@Override
	public void resetReadings() {
		numReadings = 0;
		glucoseReadings = new int[maxReadings];
	}


	@Override
	public void discardNext(int qty) {
		valuesToDiscard = qty;		
	}


	@Override
	public int getNumReadings() {
		return numReadings;
	}


	@Override
	public int getMaxReadings() {
		return maxReadings;
	}


	@Override
	public int getThreshold() {
		return threshold;
	}
	
	
	
	
	
	
}
